src/com/aca/Manager.java

```diff
@@ -0,0 +1,142 @@
+                }
+                String[] row = new String[]{
+                        gadget.getName(),
+                        String.valueOf(gadget.getDuration()),
+                        String.valueOf(gadget.getPrice()),
+                        String.valueOf(gadget.getCount()),
+                        String.valueOf(memory)
+                };
+
+                FileWriter csvWriter = new FileWriter("aca-java-june-homeworks/src/com/aca/productList/trading.csv");
+
+                    csvWriter.append(String.join(",", row));
+                    csvWriter.append("\n");
+
+
+                csvWriter.flush();
+                csvWriter.close();
+            }
+        }
+    }
+
+    public List<VideoRecorder> getAllVideoRecorders(){
+        List<VideoRecorder> recorders = new ArrayList<>();
+        for (Gadget gadget : gadgets) {
+            if (gadget instanceof VideoRecorder){
+                recorders.add((VideoRecorder)gadget);
+            }
+        }
+        return recorders;
+    }
+
+
+    public List<AudioRecorder> getAllAudioRecorders(){
+        List<AudioRecorder> recorders = new ArrayList<>();
+        for (Gadget gadget : gadgets) {
+            if (gadget instanceof AudioRecorder){
+                recorders.add((AudioRecorder)gadget);
+            }
+        }
+        return recorders;
+    }
+
+
+
+    public List<VideoRecorder> getAllVideoRecordersByType(VideoRecorder.VideoQuality videoQuality){
+        List<VideoRecorder> recorders = new ArrayList<>();
+        for (Gadget gadget : gadgets) {
+            if (gadget instanceof VideoRecorder){
+                VideoRecorder videoRecorder = (VideoRecorder) gadget;
+                if(videoRecorder.getVideoQuality() == videoQuality){
+                    recorders.add(videoRecorder);
+                }
+            }
+        }
+        return recorders;
+    }
+    public List<AudioRecorder> getAllAudioRecordersByType(AudioRecorder.AudioQuality audioQuality){
+        List<AudioRecorder> recorders = new ArrayList<>();
+        for (Gadget gadget : gadgets) {
+            if (gadget instanceof AudioRecorder){
+                AudioRecorder audioRecorder = (AudioRecorder) gadget;
+                if(audioRecorder.getAudioQuality() == audioQuality){
+                    recorders.add(audioRecorder);
+                }
+            }
+        }
+        return recorders;
+    }
+
+    public void getListOfItems(List<Gadget> gadgets){
+        for (Gadget gadget : gadgets) {
+            System.out.println(gadget.getClass()+"(count: "+gadget.getCount()+")");
+        }
+    }
+    public void changePrice(Gadget gadget, int newPrice){
+        gadget.setPrice(newPrice);
+    }
+
+    public int getSoldAmount() {
+        return soldAmount;
+    }
+
+    public void testGadgets(List<Gadget> gadgets) {
+        for (Gadget gadget : gadgets) {
+            Random random = new Random();
+            boolean onOff= random.nextBoolean();
+            if(onOff){
```

Comment:
if I call this method several times it will add items into lists several times, so clear lists in the start of method